src/couch/src/couch_file.erl

```diff
@@ -264,6 +265,19 @@ rename_file(Original) ->
         Else -> Else
     end.
 
+rename_dir(RootDelDir, Original, DbName) ->
+    DbDir = binary_to_list(DbName) ++ "_design",
+    [DbPureName | _R] = lists:reverse(filename:split(binary_to_list(DbName))),
```

Comment:
Also, while users is a Cloudant specific issue it doesn't change the logic here as if you had a database name hierarchy you'd end up just chopping off the last element for the group which seems like it'd be super confusing to any administrator trying to use file recovery. You'd basically have to write the same scripts to search for files as you would without the whole grouping effort which makes me think there's not much benefit to doing it in the first place.